src/libexpr/value/context.hh

```diff
@@ -54,10 +54,25 @@ struct NixStringContextElem {
      */
     using Built = SingleDerivedPath::Built;
 
+    /**
+     * "Poison pill" output that is rejected by `builtins.derivation`.
+     *
+     * Used to ensure the implementation of functions like
+     * `builtins.toStringDebug` do not get hashed into derivations.
+     *
+     * Encoded as ‘%poison’.
+     */
+    struct Poison {
+        bool operator ==(const Poison & other) const { return true; }
+        bool operator <(const Poison & other) const { return false; }
+        bool operator >(const Poison & other) const { return false; }
+    };
```

Comment:
Maybe add a field for an error message that explains why the string was poisoned?

Reply:
Perhaps a set of strings, considering that string contexts need to be mergeable.

Reply:
yeah also fyi this one can use operator<=> now that our c++20 story is improved.

Reply:
`operator<=>` isn't implemented for `std::set` in our C++20 🙃